src/platform/test/cinema-main.c

```diff
@@ -3,6 +3,10 @@
  * This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+#ifdef HAVE_VASPRINTF
+#define _GNU_SOURCE
+#endif
```

Comment:
This is not the correct place to define this; it should be done in CMake, and it should already be defined, though for some reason it might not be getting through to CInema.

Reply:
When building on MSYS2, the definition for vasprintf exists in libmingwex.a (automatically linked). That succeeds the linking-only verification from check_function_exists which both declares and references a function name. CInema references vasprintf but does not get a declaration because stdio.h has it behind _GNU_SOURCE.

So, I'd like to ask which impact level is preferred for other options:
1. Modify find_function to take a list of defines to keep grouped with the HAVE_ flags, for example `find_function(vasprintf _GNU_SOURCE)` appends `HAVE_VASPRINTF _GNU_SOURCE` to FUNCTION_DEFINES
2. Check for HAVE_VASPRINTF and MINGW somewhere to add the _GNU_SOURCE defintion for CInema

Reply:
You're thinking about this wrong. The problem isn't "this function needs this define on mingw", the problem is "this define needs to be set on mingw regardless", and the question is why it's set for the main binary and not CInema.

Reply:
I don't think it is, cinema-main.c is the only file with `asprintf` in any form. If you mean why the Linux CInema has it set and this is a Windows issue, it's from
https://github.com/mgba-emu/mgba/blob/7c57a5cc13c3c009916a59189d7784474aee29c0/CMakeLists.txt#L221-L223

The other information was to explain why CInema is getting the HAVE_ flag without the declaration, in other words because CheckFunctionExists.c declares it iself and only verifies linking.